### PR TITLE
fix: download profile thumbnails

### PR DESCRIPTION
**Pull request by marianogoldman**

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -60,9 +63,32 @@ export async function downloadEntityAndContentFiles(
   }
 
   const entityMetadata: {
+    type: string,
+    metadata?: any,
     content?: Array<ContentMapping>
   } = JSON.parse(contentStream)
 
+  if (entityMetadata.type === 'profile' && entityMetadata.metadata) {
```
Comment (agusaldasoro) [inline src/index.ts:71]: can we add a comment here explaining why this is needed? maybe referencing the issue?

Comment (marianogoldman) [reply src/index.ts:71]: Done

**Comment by marianogoldman** [comment]
> Add body to the files to download
Added any snapshot that can exists that is not part of the content, or eventually we might have old entities that have more items in an incomplete state.

File: src/index.ts [hunk 2]
```diff
@@ -60,9 +63,41 @@ export async function downloadEntityAndContentFiles(
...
+     *  if no need to upload them again).
+     */
+    const allAvatars: any[] = entityMetadata.metadata?.avatars ?? []
+    const snapshots = allAvatars.flatMap(avatar => Object.values(avatar.avatar.snapshots ?? {}) as string[])
+        .filter(snapshot => !!snapshot)
+        .filter(snapshot => !entityMetadata.content || entityMetadata.content.find(content => content.hash === snapshot) === undefined)
+    if (snapshots.length > 0) {
+      logger.info(`Downloading snapshots ${snapshots} for fixing entity ${JSON.stringify(entityMetadata)}`)
+      await Promise.all(
+          snapshots.map(snapshot => downloadFileWithRetries(
```
Comment (agusaldasoro) [inline src/index.ts:88]: how does this work? It takes only the values ignoring the keys?

**Review by agusaldasoro — APPROVED** [review]

File: src/index.ts (hunk 2 above)
Comment (marianogoldman) [reply src/index.ts:88]: Yes. The map looks like this:
```
          "snapshots": {
            "face": "QmWxyDrJWABXjGonFpUxJD8YYzz2xiFNxupGwYKTbySaZD",
            "face128": "QmUiEzCQPxz5eHq7KXMrGq7PiM1fnZNvZg2sWELQnuYank",
            "face256": "Qma14gteWwHrn61kv4zkRABKzopWd5ZXBppHRhbqikfaev",
            "body": "QmazJLZfUmZgNMTdwWSmJRvw4dBfcjS9GuqkwkKGRWb4K6"
          },
```

Comment (marianogoldman) [reply src/index.ts:88]: We take those ids (the values) and download the entities only if they are not included in content array.

**Pull request merged**